README.md

```diff
@@ -55,87 +47,193 @@ To set up CDF, do the following:
+
+`on_error()` function allows chart rendering even if some exception appears. It handles errors like:
+- `BAD_DOMAIN` - If bad input ranges are provided. E.g. division by zero, or sqrt of negative number
+- `OVERFLOW` - If result is more than 10^100 in absolute value
+If any of these are encountered, instead of returning a value for that timestamp, CDF returns an error field with an error message.
+To avoid these, you can wrap the (sub)expression in the on_error() function:
+```
+on_error(1/TS{externalId='canBeZero'}, 0)
+```
+
+`map()` function can handle timeseries with string values, i.e. converts strings to doubles. 
+As an example, assume that the time series "stringstate" can have values `"OPEN"` or `"CLOSED"`. Then you can convert it to a number with:
+```
+map(TS{externalId='stringstate'}, ['OPEN', 'CLOSED'], [1, 0], -1)
+```
+`"OPEN"` is mapped to 1, `"CLOSED"` to 0, and everything else to -1
 
-![Custom Query](https://raw.githubusercontent.com/cognitedata/cognite-grafana-datasource/master/images/img4.png)
+Aggregates on string timeseries is not supported right now, not even interpolation. All string time series are considered step timeseries
 
 ## Templating / Variables
 
-In order to perform templating, we enable the use of variables via `$Variable` or `[[Variable]]`.
+In order to perform templating, we enable the use of variables via `$variable` or `[[variable]]` syntax.
 
 - To add variables, go to your dashboard's settings, and then select "Variables" from the right side.
 - Make sure the "Type" is set to "Query", and then set your Cognite Data Source as the Data Source.
-- You can then specify the query to pull assets from CDF, and also filter on these assets.
+- You can then specify the query to pull assets from CDF, and also filter on these assets. For instance:
+```
+assets{parentIds=[123], name=~"test-.*"}
+```
+Query above requests for assets with parameter `parentIds=[123]` and filter results by `name` matches `test-.*` regexp
 
-![Variable Query](https://raw.githubusercontent.com/cognitedata/cognite-grafana-datasource/master/images/img7.png)
+Full list of params is available in [CDF documentation](https://docs.cognite.com/api/v1/#operation/listAssets) 
 
-![Variable as root asset](https://raw.githubusercontent.com/cognitedata/cognite-grafana-datasource/master/images/img8.png)
+![Variable Query](https://raw.githubusercontent.com/cognitedata/cognite-grafana-datasource/docs/dashboard-development/images/readme/image5.png)
+
+![Variable as root asset](https://raw.githubusercontent.com/cognitedata/cognite-grafana-datasource/docs/dashboard-development/images/readme/image6.png)
+
+Variables can be also formatted in way to fit it into the query, for instance in case when you have multivalue variable:
+```
+ts{assetIds=[${variable:csv}]}
+```
+Variable will be parsed to comma separated value in case if few assets will be selected from the dropdown.
```

Comment:
```suggestion
Variable will be serialised to comma separated value in case if few assets will be selected from the dropdown.
```
can use `composed` as well